src/Streamly/Internal/Data/Stream/Async.hs

```diff
@@ -784,21 +814,27 @@ consMWAsync m r = fromStream $ K.yieldM m `wAsync` (toStream r)
 -- @since 0.2.0
 {-# INLINE wAsync #-}
 wAsync :: (IsStream t, MonadAsync m) => t m a -> t m a -> t m a
-wAsync = joinStreamVarAsync WAsyncVar
-
--- | The 'Semigroup' operation for 'WAsyncT' interleaves the elements from the
--- two streams.  Therefore, when @a <> b@ is evaluated, one action is picked
--- from stream @a@ for evaluation and then the next action is picked from
--- stream @b@ and then the next action is again picked from stream @a@, going
--- around in a round-robin fashion. Many such actions are executed concurrently
--- depending on 'maxThreads' and 'maxBuffer' limits. Results are served to the
--- consumer in the order completion of the actions.
---
--- Note that when multiple actions are combined like @a <> b <> c ... <> z@ we
--- go in a round-robin fasion across all of them picking one action from each
--- up to @z@ and then come back to @a@.  Note that this operation cannot be
--- used to fold a container of infinite streams as the state that it needs to
--- maintain is proportional to the number of streams.
+wAsync = joinStreamVarWAsync WAsyncVar
+
+-- | The 'Semigroup' operation (@<>@) for 'WAsyncT' merges two streams
+-- concurrently with fair scheduling of actions from both the streams i.e.
+-- actions from each stream are scheduled for execution in a round robin
+-- fashion.  The scheduled actions may be executed concurrently and the results
+-- are consumed in the order in which they become available.  In @s1 <> s2 <>
+-- s3 ...@ action from the streams s1, s2 and s3 are scheduled for execution in
+-- a round-robin fashion.
+--
+-- 'WAsyncT' is similar to 'WSerialT' but with concurrent execution.  However,
+-- 'WAsyncT' is different from 'WSerialT' in the manner in which @a <> b <> c@
+-- is scheduled.  'WAsyncT' schedules all the streams fairly whereas 'WSerialT'
+-- is skewed towards the streams on the left.  See notes in 'WSerialT' for
```

Comment:
Here it says, 
>`WSerialT` is skewed towards the streams on the left

and in the documentation for `WSerialT` it says
> `WSerialT` is skewed towards the elements on the right.

Reply:
Right is wrong, left is right!